File: client/src/app/shared/ResourseSpecification.ts
import { CompositionItem } from "./CompositionItem";
import { DishValue } from "./DishValue";

export class ResourseSpecification {

    id: number = 0;
    composition: CompositionItem[] = [];
    outputDishWeightG: number = 0;
    dishValue: DishValue = new DishValue;
    
}

File: Data/Entities/ResourseSpecification.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FoodDiary.Data.Entities
{
    public class ResourseSpecification
    {
        public int Id { get; set; }
        
        public Dish Dish { get; set; }
        public int DishId { get; set; }

        public ICollection<CompositionItem> Composition { get; set; }

        public double OutputDishWeightG { get; set; }

        public DishValue DishValue { get; set; }

        public int DishValueId { get; set; }



    }
}
